### PR TITLE
Add/plugin configurator

### PR DESCRIPTION
**Pull request by jeffersonrabb**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Introduces the `Configuration_Manager` abstract class and two subclasses, for AMP and Jetpack. Ultimately, there will be a subclass for every plugin that requires any configuration. Currently the AMP Configuration Manager switches to AMP strict mode when `configure()` is called. The Jetpack Configuration Manager shows how `is_configured()` can be used to determine if Jetpack has been connected. 

This also introduces a new endpoint: `newspack/v1/plugins/{plugin-slug}/configure`. This endpoint will install and activate the plugin, then if a configuration class has been specified it will be loaded and `configure()` called. If the plugin is already active, this endpoint won't fail like `/activate` does. Instead the error will be ignored and the plugin configured normally.

### How to test the changes in this Pull Request:

Testing can be a little tricky. In one approach, you could throw some PHP somewhere and try the calls out. This might look like this:

```
require_once NEWSPACK_ABSPATH . '/includes/configuration_managers/class-jetpack-configuration-manager.php'; 
$config_manager = new Jetpack_Configuration_Manager();
if ( $config_manager->is_configured() ) { 
  echo '<h1>Jetpack Configured</h1>';
}
```

The `<h1>` will be shown only if Jetpack is installed and connected.

Alternatively, edit the `PluginInstaller` component and change the `activate` endpoint to `configure` here: https://github.com/Automattic/newspack-plugin/blob/master/assets/components/src/plugin-installer/index.js#L78. Run the build script, then go to the Components Demo and try installing AMP. To verify the change, go into the AMP dashboard and switch the mode to Reader or Transitional. Then delete AMP. Go to Components Demo and install AMP. It should  install and activate normally, and after this is done you should see that the mode has been changed to Standard.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: includes/configuration_managers/class-jetpack-configuration-manager.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	 */
+	public $slug = 'jetpack';
+
+	/**
+	 * Get this wizard's name.
+	 *
+	 * @return string The wizard name.
+	 */
+	public function configure() {
+		return true;
```
Comment (claudiulodro) [inline includes/configuration_managers/class-jetpack-configuration-manager.php]: Nitpicks: The docblock for `configure` in the abstract class says it shouldn't return anything. Also the docblock for this method is clearly from some other method. :)

File: includes/configuration_managers/class-configuration-manager.php [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+	/**
+	 * Retrieve data for this plugin.
+	 *
+	 * @var object
+	 */
+	protected function get_plugin_data() {
+		$managed_plugins = Plugin_Manager::get_managed_plugins();
+		try {
+			$plugin = $managed_plugins[ $this->slug ];
+		} catch ( Exception $e ) {
```
[the remark this answers is not in the document]
Comment (jeffersonrabb) [reply includes/configuration_managers/class-configuration-manager.php]: Reworked this in https://github.com/Automattic/newspack-plugin/pull/126/commits/6791213f9057dd7beb22eeb13fcd54bb02742fca

File: includes/configuration_managers/class-jetpack-configuration-manager.php (hunk 1 above)
Comment (jeffersonrabb) [reply includes/configuration_managers/class-jetpack-configuration-manager.php]: Sloppy, sloppy... https://github.com/Automattic/newspack-plugin/pull/126/commits/ad64293d40396d7e251d9a4411e1e4137fe3f398

**Comment by jeffersonrabb** [comment]
Great, thank you!

**Pull request merged**